pkg/controller/command/verifiable/command.go

```diff
@@ -245,3 +250,46 @@ func (o *Command) GetCredentials(rw io.Writer, req io.Reader) command.Error {
 
 	return nil
 }
+
+// GeneratePresentation generates verifiable presentation from a verifiable credential.
+func (o *Command) GeneratePresentation(rw io.Writer, req io.Reader) command.Error {
+	request := &Credential{}
+
+	err := json.NewDecoder(req).Decode(&request)
+	if err != nil {
+		logutil.LogInfo(logger, commandName, generatePresentationCommandMethod, "request decode : "+err.Error())
+
+		return command.NewValidationError(InvalidRequestErrorCode, fmt.Errorf("request decode : %w", err))
+	}
+
+	// TODO https://github.com/hyperledger/aries-framework-go/issues/1316 Add keys for proof
+	//  verification as options to the function.
+	vc, _, err := verifiable.NewCredential([]byte(request.VC))
```

Comment:
TODOs:

- VP consists of 1 or more VCs (meaning the VC parameter can be an array).
- It might be nice to have an option to use one or more stored VCs by ID.
- VP can be extended with additional JSON-LD contexts. This API currently doesn't support anything but the base VP context.

Reply:
Created an issue for these TODOs - https://github.com/hyperledger/aries-framework-go/issues/1421